viscy/translation/engine.py

```diff
@@ -530,6 +530,59 @@ def __init__(
     def forward(self, x: Tensor) -> Tensor:
         return self.model(x)
 
+    @torch.no_grad()
+    def inference_tiled(
+        self, x: torch.Tensor, out_channel: int = 2, step: int = 1
+    ) -> torch.Tensor:
+        """
+        Example:
+            pred = VS_inference_t2t(input_tensor, config)
+            # input_tensor: torch.Tensor of shape (B, 1, Z, Y, X)
+            # pred: torch.Tensor of shape (B, 2, Z, Y, X)
+        """
+
+        self.eval()
+        assert x.ndim == 5, f"Expected shape (B,C,Z,Y,X), got {x.shape}"
+
+        B, _, Z, Y, X = x.shape
+        in_stack_depth = self.model.out_stack_depth
+
+        out_tensor = x.new_zeros((B, out_channel, Z, Y, X))
+        weights = x.new_zeros((1, 1, Z, 1, 1))
+
+        pad = getattr(self, "_predict_pad", None)
```

Comment:
Use a normal conditional. In general, avoid string-based dynamic programming as much as possible.